### DISCUSSION TITLE
[Discuss] Obsidian and Docusaurus

### DISCUSSION THREAD

**Discussion by andrey-jef** [anaclumos/www#36] [General]
Hi,

Just wanna drop by, and say hello.

For the context, I was searching how to publish an Obsidian vault using Docusaurus template. Then I lurked into Obsidian discord, then Docusaurus channel, and bump, I found your blog. Really a beautiful gem.

I share [your vision about digital pkm](https://cho.sh/w/D8FB8E), especially the "nameless routes" part. People argue that it could affect SEO, but personally I don't care much yet about SEO with a personal digital garden. Your unique-id approach is similar to Dendron. They even plan to implement the url slug like Notion, that is "the-long-name-can-be-changed-$unique-id$", where "the-long-name" part (easier for human to read) can be updated accordingly to the change of filename, but the route is parsed by only the "unique-id" part. Sadly, Dendron dev have recently decided to move Dendron project into maintenance mode due to business decision.

When you have interest to write a tutorial how to replicate your publishing workflow with Obsidian and Docusaurus, please count me as one enthusiast reader, I'd love to learn your path.

Thanks

**Comment by anaclumos** [comment]
Hey, thanks for reaching out! I did not know dendron went into maintenance mode; indeed PKMs are tough businesses in a way that everyone's workflow is different, and thus very challenging to find the Product-Market Fit 😭

I use nameless routes mainly because I change the title of a post **a lot** and I want links to stay intact after updating the post title. Thus, I randomly assign an irrelevant hex digit for each article. Surprisingly, I did not feel any harm in SEO, given that my sitemap.xml faithfully shows all links. I constantly get 1K+ clicks from Google every month (comparable to my previous https://github.com/TryGhost/ghost blog.)

Docusaurus is not the perfect solution, tho. I want every hex to be top-level; for example, I want `cho.sh/D8FB8E` instead of the current `cho.sh/w/D8FB8E`. This is only possible by rewriting a significant portion of the docusaurus; right now doing this will cause a collision between the Docs plugin and the Blog plugin. Some other limitations of Docusaurus

- No first-level Wikilink supports
- No first-level Backlinking & Graph supports
- No first-level `figcaption` supports
  - I made a [janky implementation](https://github.com/anaclumos/extracranial/blob/main/tools/img-alt-to-figcaption.py), which subsequently broke img optimizations
  - So, no blurry image loadings as of right now (in the sacrifice of Wikilink and figcaption, you can use the Docusaurus Ideal Image plugin)
- No Excalidraw
- No Footnote
- No automatic (or assisted) insights

I may try to rebuild everything from scratch at one point. I'd make a meta-meta-framework like Nextra based on all the modern features of Next.js, plus optimize for Memex with the features mentioned above. Previous builds of https://github.com/junhoyeo/paracosm had a Memex incorporated but then later removed. [Screenshot from when it had a Memex.](https://cho.sh/r/AB0F3F)

CC-ing @stevejkang @junhoyeo

**Comment by anaclumos** [comment]
Also, if you simply want the high-level functionalities of this website, try [Mem](https://get.mem.ai/); they have really good software.

**Comment by seiwonpark** [comment]
@anaclumos (Sorry that I'm not sure if this comment is related to the issue)

What a nice {wiki | blog | brain} I've ever seen! Likewise, I've been also working on building my own wiki-like page. And I really liked your **Graph** idea that each node links one another to make some kind of relation between them. So I've tried applying this feature to my page but here I've got a question.

Q. How do you manage links that previously didn't have any links but after adding some posts there seems to be some relation between them? 

For example, I posted about "Docusaurus 1.0" yesterday and if I'm trying to post about "Docusaurus 2.0", do you modify the post of "Docusaurus 1.0" as well? (I mean by adding some links to "Docusaurus 2.0") Or do you just add link from "Docusaurus 2.0" to "Docusaurus 1.0"?

Anyway thanks for sharing your cool "Extracranial".

**Comment by anaclumos** [comment]
Hey @SeiwonPark, thank you so much for your interest!

In the example you mentioned, you will have to add a link manually. Some tools help me to do this, such as [Obsidian Ava](https://app.anotherai.co/), but these days I do it myself.

Not the best design 😃 I hope the next-gen tools do this automatically. Examples include [Saga](https://saga.so)'s Autolinking or [Mem](https://get.mem.ai/)'s 'Similar to this Mem' feature, but none of them are perfect.

There's also the problem of Remem (coined in Exhalation by Chiang, 숨, 테드 창 저) -- if the machine does everything for me, my brain will slowly degrade over time. Perhaps leaving some chores for my brain is an excellent middle ground.

**Comment by anaclumos** [comment]
Oh. Also, I misunderstood quite a bit. If you add one link, i.e., docusaurus 2.0 → 1.0, you don't need to add the opposite. It will automatically appear as a backlink.

What I meant by linking myself is the initial [[]] bracket connecting process.

**Comment by seiwonpark** [comment]
Thank you for the quick response!

> if the machine does everything for me, my brain will slowly degrade over time. Perhaps leaving some chores for my brain is an excellent middle ground.

I really appreciate your insight :) 

<br/>   

> It will automatically appear as a backlink.

> What I meant by linking myself is the initial [[]] bracket connecting process.

Ohhh sorry for misleading. I actually meant both of them. So it would automatically generate links by `backlink`! But still it needs to be added manually for the `hyperlink([ ])`. And you're willing to do some chores for your brain 👍  How cool!

You're giving so much impression to me and others as well. Thanks for your work!
